client/state/current-user/reducer.js

```diff
@@ -82,9 +90,28 @@ export const capabilities = createReducer( {}, {
 	}
 }, capabilitiesSchema );
 
+export const gravatarStatus = createReducer( {}, {
+	[ GRAVATAR_UPLOAD_START ]: state => {
+		return Object.assign( {}, state, {
+			isUploading: true
+		} );
+	},
+	[ GRAVATAR_UPLOAD_SUCCESS ]: state => {
+		return Object.assign( {}, state, {
+			isUploading: false
+		} );
+	},
+	[ GRAVATAR_UPLOAD_FAILURE ]: state => {
+		return Object.assign( {}, state, {
+			isUploading: false
+		} );
+	},
+}, gravatarSchema );
```

Comment:
> Remove the max-age

Let's try to avoid this one. Max-age is currently only set to 300.

> Instead, we could store the cache-busting URL directly on state.users.items.[id].avatar_URL and we would never have to change it back to the original URL.

Interesting idea. What happens when we refresh user data? The current user is set once per page load, so what happens when we refresh the page within that 5 min cache period? (Maybe we don't care?)

Here's another case: say we have our browser open for more than 5 minutes. How do we remove the local image from the redux store?

> 1. What happens if a user is using a Photon URL (there's a reference to it in <Gravatar /> and getResizedImageURL())

Huh didn't notice that. Does anyone have steps to recreate this case? 

> Are we always passing the current user object from the state to Gravatar? I think so, but will need to verify

This is a bit of an aside, but I would suggest refactoring `<Gravatar />` to be a little dumber, and take in the displayName and avatarUrl to be explicit props instead of passing in a full user object. Not sure if it fits in scope here, but we may want to consider it if there aren't too many usages.